api/messages/publish.proto

```diff
@@ -82,4 +82,19 @@ message PublishReply {
  // The highest sequential index that has been persisted. (See the API
  // README for details on what "persisted" entails.)
  uint64 persisted_index = 4;
+
+ // The list of event rejections.
+ // An event might not pass the validation,
+ // e.g. does not have a required field or has invalid metadata or fields
+ repeated Rejection rejected = 5;
+}
+
+// Rejection is a status object that describes why a certain
+// event from the batch was rejected by the server.
+message Rejection {
+ // Index of the rejected event in the initial `events` list.
+ uint64 index = 1;
+
+ // Human-readable message that explains the reason of the rejection
+ string reason = 2;
```

Comment:
Making this `google.rpc.Status` will let us include an error code (`InvalidArgument`) in addition to an error message. 

It also gives us flexibility to report more structured errors here, as you can embed other protobuf messages as a payload if we ever wanted to do that. This could be nice for processing errors in the future, where there may be a chain of errors or something like a stack trace we want to include.

Reply:
For processor errors we won't necessarily know those at the point where we publish to the queue though.